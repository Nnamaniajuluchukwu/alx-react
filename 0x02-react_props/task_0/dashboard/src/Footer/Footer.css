:root {
    --font-family-general: Arial, Helvetica, sans-serif;
}

.App-footer {
    text-align: center;
    padding: 1rem;
    font-style: italic;
    font-family: var(--font-family-general);
}
