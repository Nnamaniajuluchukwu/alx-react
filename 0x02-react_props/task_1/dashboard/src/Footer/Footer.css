.App-footer {
    text-align: center;
    padding: 1rem;
    font-style: italic;
    font-family: var(--font-family-general);
}
