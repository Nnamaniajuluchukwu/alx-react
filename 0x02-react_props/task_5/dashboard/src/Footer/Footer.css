.App-footer {
    text-align: center;
    padding: 1rem;
    font-style: italic;
    font-family: var(--font-family-general);
    position: fixed;
    bottom: 0;
    border-top: 3px solid #e0354b;
    width: 100%
}
