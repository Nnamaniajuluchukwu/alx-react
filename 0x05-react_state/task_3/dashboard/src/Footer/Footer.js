import React, { useContext } from "react";
import { getFullYear, getFooterCopy } from '../utils/utils';
import AppContext from '../App/AppContext';


export default function Footer() {
    const { user } = useContext(AppContext);
    return (
        <footer className='App-footer'>
            Copyright {getFullYear()} - {getFooterCopy(true)}
            { user.isLoggedIn &&
                <p><a href="##">Contact us</a></p>
            }
        </footer>
    );
}
