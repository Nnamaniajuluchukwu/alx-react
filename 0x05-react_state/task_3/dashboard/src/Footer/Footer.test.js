import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Footer from './Footer';
import AppContext from '../App/AppContext';


describe.skip('Footer Component', () => {
    let wrapper;

    const contextValue = {
        user: {
          email: '',
          password: '',
          isLoggedIn: false
        },
        logOut: () => {}
    };

    beforeEach(()=>{
        wrapper = mount(
            <AppContext.Provider value={contextValue}>
                <Footer />
            </AppContext.Provider>
        );
    });

    it('should render without crashing', () => {
        expect(wrapper.exists()).toBeTruthy();
    });

    describe('Footer Component', () => {
        it('should render the text "Copyright"', () => {
          expect(wrapper.text()).toContain('Copyright');
        });
    });

    it('should match the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
