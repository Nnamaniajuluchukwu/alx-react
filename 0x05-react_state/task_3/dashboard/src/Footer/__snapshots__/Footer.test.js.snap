// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match the snapshot 1`] = `<Footer />`;
